### PR TITLE
AddTemporalIdsToInterpolationTarget now works with time-dependent maps

### PR DESCRIPTION
**Pull request by markscheel**
## Proposed changes
There are multiple commits in this PR. Most of them do small things:
 - Add 'frame' type alias to InterpolationTargets
 - Add a new tag PendingTemporalIds to the DataBox of InterpolationTarget
 - Adds new functions to InterpolationTargetDetail, for determining when maps are time-dependent.
And then some make larger changes
 - Add intrp::Actions::VerifyTemporalIdsAndSendPoints (as described in the documentation)
 - Update AddTemporalIdsToInterpolationTarget, and its test, to work with time-dependent maps.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Comment by markscheel** [comment]
Rebased. No longer dependent on another PR.

File: src/NumericalAlgorithms/Interpolation/Actions/VerifyTemporalIdsAndSendPoints.hpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    Parallel::GlobalCache<Metavariables>& cache) noexcept {
+  using TemporalId = typename Metavariables::temporal_id::type;
+
+  const auto& pending_temporal_ids =
+      db::get<Tags::PendingTemporalIds<TemporalId>>(*box);
+  if (pending_temporal_ids.empty()) {
+    return; // Nothing to do if there are no pending temporal_ids.
+  }
+
+  std::vector<TemporalId> new_temporal_ids{};
```
Comment (wthrowe) [inline src/NumericalAlgorithms/Interpolation/Actions/VerifyTemporalIdsAndSendPoints.hpp]: Move this down to where it is used.

File: src/NumericalAlgorithms/Interpolation/InterpolationTargetDetail.hpp [hunk 2]
```diff
@@ -277,6 +277,23 @@ using cache_contains_functions_of_time =
...
+template <typename InterpolationTargetTag, typename DbTags,
+          typename Metavariables>
+bool maps_are_time_dependent(
+    const db::DataBox<DbTags>& box,
+    const tmpl::type_<Metavariables>& /*meta*/) noexcept {
+  const auto& domain =
+      db::get<domain::Tags::Domain<Metavariables::volume_dim>>(box);
+  for (const auto& block : domain.blocks()) {
+    if (block.is_time_dependent()) {
+      return true;
```
Comment (wthrowe) [inline src/NumericalAlgorithms/Interpolation/InterpolationTargetDetail.hpp]: Probably clearer as `return alg::any_of(domain.blocks(), [](const auto& block) noexcept { return block.is_time_dependent(); }))`.

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_AddTemporalIdsToInterpolationTarget.cpp [hunk 3]
```diff
@@ -144,12 +162,56 @@ void test_add_temporal_ids() {
...
+            target_component, ::intrp::Tags::TemporalIds<temporal_id_type>>(
+            runner, 0)
+            .empty());
+
+  // At this point, there should be only one queued simple action,
+  // VerifyTemporalIdsAndSendPoints, and triggering this action
+  // should move the PendingTemporalIds to TemporalIds and invoke
+  // other simple actions.
+  CHECK(runner.template number_of_queued_simple_actions<target_component>(0) ==
+        1);
```
Comment (wthrowe) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_AddTemporalIdsToInterpolationTarget.cpp]: Use the ActionTesting::* helper functions (several places).

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_AddTemporalIdsToInterpolationTarget.cpp [hunk 4]
```diff
@@ -215,10 +294,315 @@ void test_add_temporal_ids() {
...
+                                typename metavars::InterpolationTargetA>;
+
+  const double expiration_time = 0.1;
+  // Create a Domain with time-dependence. For this test we don't care
+  // what the Domain actually is, we care only that it has time-dependence.
+  const auto domain_creator = domain::creators::Brick(
+      {{-1.2, 3.0, 2.5}}, {{0.8, 5.0, 3.0}}, {{1, 1, 1}}, {{5, 4, 3}},
+      {{false, false, false}},
+      std::unique_ptr<domain::creators::time_dependence::TimeDependence<3>>(
+          new domain::creators::time_dependence::UniformTranslation<3>(
```
Comment (wthrowe) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_AddTemporalIdsToInterpolationTarget.cpp]: Use make_unique.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by markscheel** [comment]
Made fixups.

**Comment by wthrowe** [comment]
Squash.

**Comment by markscheel** [comment]
Squashed (and rebased).

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
@wthrowe this was squashed and rebased

**Review by wthrowe — APPROVED** [review]

**Comment by nilsvu** [comment]
@markscheel Could you update the upgrade instructions in the PR description, so they give instructions what to do when rebasing or delete them if nothing must be done?

**Comment by nilsvu** [comment]
@markscheel I deleted the upgrade instructions that said "do nothing" because I'm preparing the release now. Please add upgrade instructions if you think they would be useful here.

**Pull request merged**